src/harness/reference_models/iap/iap_example.py

```diff
@@ -0,0 +1,76 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
```

Comment:
example code is nice but of limited utility. It only test that the code runs without crashing. 
Can you put a real set of unit test ?
You can easily mock or fake the interference routines, which are the only external routine you need, and then compare the behavior of the algorithm as expected, for simple test cases

Reply:
we will work on this later.